pip_audit/_dependency_source/pyproject.py

```diff
@@ -76,21 +76,26 @@ def collect(self) -> Iterator[Dependency]:
             # dependency resolution now, we can think about doing `pip install <local-project-dir>`
             # regardless of whether the project has a `pyproject.toml` or not. And if it doesn't
             # have a `pyproject.toml`, we can raise an error if the user provides `--fix`.
-            with NamedTemporaryFile() as req_file, TemporaryDirectory() as ve_dir:
-                # Write the dependencies to a temporary requirements file.
-                req_file.write(os.linesep.join(deps).encode())
-                req_file.flush()
-
-                # Try to install the generated requirements file.
-                ve = VirtualEnv(install_args=["-r", req_file.name], state=self.state)
-                try:
-                    ve.create(ve_dir)
-                except VirtualEnvError as exc:
-                    raise PyProjectSourceError(str(exc)) from exc
-
-                # Now query the installed packages.
-                for name, version in ve.installed_packages:
-                    yield ResolvedDependency(name=name, version=version)
+            with TemporaryDirectory() as ve_dir:
+                with NamedTemporaryFile(dir=ve_dir, delete=False) as req_file:
```

Comment:
Nitpick: use a chained `with` statement to make this terser:

```suggestion
            with TemporaryDirectory() as td, NamedTemporaryFile(dir=td, delete=False) as req_file:
```